src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -588,6 +591,82 @@ public void availableNoticeIsSentOnceWhenItemStatusIsChangedToAwaitingPickup() {
       .until(patronNoticesClient::getAll, empty());
   }
 
+  @Test
+  public void accountRecordIsCreatedWhenItemIsOverdue() throws InterruptedException {
+    useFallbackPolicies(loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.activeNotice().getId(),
+      overdueFinePoliciesFixture.facultyStandardDoNotCountClosed().getId(),
+      lostItemFeePoliciesFixture.facultyStandard().getId());
+
+    final IndividualResource james = usersFixture.james();
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      item -> item.withPrimaryServicePoint(checkInServicePointId));
+    final IndividualResource nod = itemsFixture.basedUponNod(item ->
+      item.withPermanentLocation(homeLocation.getId()));
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(nod, james,
+      new DateTime(2020, 1, 1, 12, 0, 0, DateTimeZone.UTC));
+
+    JsonObject servicePointOwner = new JsonObject();
+    servicePointOwner.put("value", homeLocation.getJson().getString("primaryServicePoint"));
+    servicePointOwner.put("label", "label");
+    UUID ownerId = UUID.randomUUID();
+    feeFineOwnersClient.create(new FeeFineOwnerBuilder()
+      .withId(ownerId)
+      .withOwner("fee-fine-owner")
+      .withServicePointOwner(Collections.singletonList(servicePointOwner))
+    );
+
+    UUID feeFineId = UUID.randomUUID();
+    feeFinesClient.create(new FeeFineBuilder()
+      .withId(feeFineId)
+      .withFeeFineType("Overdue fine")
+      .withOwnerId(ownerId)
+    );
+
+    loansFixture.checkInByBarcode(new CheckInByBarcodeRequestBuilder()
+        .forItem(nod)
+        .on(new DateTime(2020, 1, 25, 12, 0, 0, DateTimeZone.UTC))
+        .at(checkInServicePointId));
+
+    TimeUnit.SECONDS.sleep(1);
```

Comment:
If the tests need to wait for a condition, rather than using a sleep in the tests, please use Awaitility instead to do eager polling of the condition

Reply:
Done